src/common/types/string_type.cpp

```diff
@@ -25,4 +26,100 @@ void string_t::Verify() const {
 	}
 }
 
+string_t string_t::operator>>(const idx_t &shift) const {
```

Comment:
Could we perhaps implement these as methods in the `Bit` class as well? They only make sense if the `string_t` is a `Bit`.

Reply:
I’ve been struggling with where to place these operator overloads and I agree the bit class would be much better. However because the internal type of BIT is just a string_t, I’m not sure how I could overload these solely on the BIT type. I’ve now pushed two possible solutions. For the left and right shift I’ve made static functions that get called from their template specialisations in bitwise.cpp. For the other operators, I’ve just moved the function bodies to bit.cpp, but they are still overloaded on string_t. Would either of these solutions be ok? Or do you maybe have another suggestion for me?

Reply:
Can we avoid overloading `string_t` entirely and instead call them as methods, e.g. `Bit::LeftShift`, `Bit::RightShift`, etc?

This might require a bit of extra work for the bitwise functions (e.g. `BitwiseANDOperator` has to be overloaded for bits to call this function instead of using the operator) - but I think it is worth it to not pollute the `string_t` class with this.

Reply:
I think, I've fixed it now!